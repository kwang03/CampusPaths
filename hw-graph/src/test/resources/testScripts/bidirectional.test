#Tests the functionality of having bidirectional edges in our graph

#Create graph
CreateGraph myGraph

#Adds some nodes to the graph
AddNode myGraph n1
AddNode myGraph n2
AddNode myGraph n3
AddNode myGraph n4

#Adds a bidirectional edge between two nodes with the same label
AddEdge myGraph n1 n2 e
AddEdge myGraph n2 n1 e
ListChildren n1
ListChildren n2

#Adds a bidirectional edge between two nodes with different labels
AddEdge myGraph n3 n4 e34
AddEdge myGraph n4 n3 e43
ListChildren n3
ListChildren n4
