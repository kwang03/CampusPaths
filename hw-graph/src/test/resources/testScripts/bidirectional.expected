#Tests the functionality of having bidirectional edges in our graph

#Create graph
created graph myGraph

#Adds some nodes to the graph
added node n1 to myGraph
added node n2 to myGraph
added node n3 to myGraph
added node n4 to myGraph

#Adds a bidirectional edge between two nodes with the same label
added edge e from n1 to n2 in myGraph
added edge e from n2 to n1 in myGraph
the children of n1 in myGraph are: n2(e)
the children of n2 in myGraph are: n1(e)

#Adds a bidirectional edge between two nodes with different labels
added edge e34 from n3 to n4 in myGraph
added edge e43 from n4 to n3 in myGraph
the children of n3 in myGraph are: n4(e34)
the children of n4 in myGraph are: n3(e43)